File: 6KYU/Calculate Pyramid Height/solution.go
package kata

func PyramidHeight(n int) int {
	var i = 0

	for {
		i = i + 1

		n = n - (i * i)

		if n < 0 {
			return i - 1
		}

		if n == 0 {
			return i
		}
	}

	return i
}

File: 6KYU/Calculate Pyramid Height/solution.cs
using System;
public class Kata
{
  public static int PyramidHeight (int n)
  {
    var i = 0;
    do
    {
        i++;
        n -= i * i;
        if (n < 0)
            i--;

    } while (n > i * i);

    return i;
   }
}